lib/src/short/line_splitting/solve_state.dart

```diff
@@ -525,7 +525,7 @@ class SolveState {
     var unboundConstraints = <Rule, Set<int>>{};
     for (var unbound in _unboundRules) {
       // Lazily create and add the set to the constraints only if needed.
-      late final disallowedValues = unboundConstraints[unbound] = {};
+      late var disallowedValues = unboundConstraints[unbound] = {};
```

Comment:
Tricky. I like.
(But ti's only used in one place, so I'd avoid the complexity of understanding `late` and when its expression is evaluated, and just  do:
```dart
  Set<int>? disallowedValues;
  ....
     (disallowedValues ??= unboundConstraints[unbound] = {}).add(value)
```
It's the same thing, just more direct.
Maybe I just don't like `late`. Yep, that's it.)

Reply:
This is in the old formatter code, so I'm trying not to touch it too much since it will go away at some point when we no longer support the old style. The only reason there's a change here is because I turned on the "unnecessary_file" lint to be more consistent about using `var` for locals.

But, I agree, using `late` for a local is always a little funny feeling. But in this case, laziness is exactly what I want and the language does support it, so I figured I may as well use it.